### PR TITLE
feat(flow): update assessment flows to 9a (#87)

### PR DESCRIPTION
**Pull request by NetBKarine**
## Description
Added self-isolation mini-form in assessment form. This necessitated to make "subforms" of both to include in three now separated assessment forms. For inheritance reasons these mini-forms could not be FormActions or have the same methods, except for validation methods to pass in inheritance.

This also meant including contact and travel slots in the form, even if they are only used in the basic assessment

Unit tests for the assessment form were replaced by tests for the subform because asjusting them for the 3 flows was too complex to have something fast enough, for little gain

*Lacks french translations - should not be merged yet*

## Checklist

- [ X] PR title follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) guidelines 
- [ X] All relevant PR sections are populated, irrelevant ones are removed

### CODE REVIEW

File: action-server/actions/assessment_common.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            risk_level_value.append("elevated-covid-risk")
+
+        return [
+            SlotSet(
+                AssessmentSlots.RISK_LEVEL,
+                "common" if risk_level_value == [] else ",".join(risk_level_value),
+            ),
+            SlotSet(AssessmentSlots.SYMPTOMS, symptoms_value),
+            SlotSet(AssessmentSlots.SELF_ASSESS_DONE, True),
+        ]
```
Comment (LaurenceDupont) [inline action-server/actions/assessment_common.py:148]: We should extract some logic in functions

File: core/domain/domain.fr.yml [hunk 2]
```diff
@@ -204,8 +192,12 @@ responses:
...
           payload: "/less"
 
-  utter_maybe_cured:
-    - text: "Il se peut que vous soyez guéri(e). Nous vous invitons à rester à la maison et suivre les consignes de distanciation sociale. Si vous sortez, je vous recommande fortement de porter un masque."
+  # Next two to be translated
+  utter_maybe_recovered:
+    - text: "You might have recovered. I invite you to stay home as much as possible and comply with social distancing rules. If you go outside, I recommend that you wear a mask."
+
+  utter_more_information:
+    - text: "For more information, resources and recommendations about Covid-19, please visit [covid19.dialogue.co](https://covid19.dialogue.co/#/info?id=common)"
```
[the remark this answers is not in the document]
Comment (NetBKarine) [reply core/domain/domain.fr.yml]: I thought I had pushed the addition of french messages (they were finished after I submitted the first time). They're all available now.

File: action-server/actions/assessment_common.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+
+class AssessmentCommon:
+    @staticmethod
+    def base_required_slots(tracker: Tracker) -> List[Text]:
+        slots = [AssessmentSlots.SEVERE_SYMPTOMS]
+
+        if tracker.get_slot(AssessmentSlots.SEVERE_SYMPTOMS) is False:
+            slots += [
+                AssessmentSlots.SEVERE_SYMPTOMS,
```
Comment (LaurenceDupont) [inline action-server/actions/assessment_common.py:43]: You are adding the severe symptoms slot twice

File: action-server/actions/checkin_return_form.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+        return await super()._activate_if_required(dispatcher, tracker, domain)
+
+    @staticmethod
+    def required_slots(tracker: Tracker) -> List[Text]:
+        slots = [AssessmentSlots.SEVERE_SYMPTOMS]
+
+        if tracker.get_slot(AssessmentSlots.SEVERE_SYMPTOMS) is False:
+            slots += [
+                AssessmentSlots.SEVERE_SYMPTOMS,
```
Comment (LaurenceDupont) [inline action-server/actions/checkin_return_form.py:34]: We are adding the severe symptoms slot twice

**Pull request merged**